### PR TITLE
[Merged by Bors] - fix(Topology/Order/UpperLowerSetTopology): Fix theorem names in UpperLowerSetTopology to match mathlib4 conventions

### PR DESCRIPTION
**Pull request by mans0954**
Modify lemma names in `UpperLowerSetTopology` to meet Mathlib4 conventions.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Order/UpperLowerSetTopology.lean [hunk 1]
```diff
@@ -285,7 +284,7 @@ theorem IsOpen_sInter {S : Set (Set α)} (hf : ∀ s ∈ S, IsOpen s) : IsOpen (
   exact hf _ hs
 
 -- Alexandrov property, index formulation
-theorem isOpen_iInter {f : ι → Set α} (hf : ∀ i, IsOpen (f i)) : IsOpen (⋂ i, f i) := by
+theorem IsOpen_iInter {f : ι → Set α} (hf : ∀ i, IsOpen (f i)) : IsOpen (⋂ i, f i) := by
```
Comment (eric-wieser) [inline Mathlib/Topology/Order/UpperLowerSetTopology.lean:287]: This new name is wrong, theorems should be `camelCase_with_underscores`

File: Mathlib/Topology/Order/UpperLowerSetTopology.lean [hunk 2]
```diff
@@ -26,8 +26,7 @@ topology does not coincide with the lower topology.
 - `UpperSetTopology.IsOpen_iInter` - the intersection of any indexed collection of open sets is open
```
Comment (eric-wieser) [inline Mathlib/Topology/Order/UpperLowerSetTopology.lean:26]:
```suggestion
- `UpperSetTopology.isOpen_iInter` - the intersection of any indexed collection of open sets is open
```

File: Mathlib/Topology/Order/UpperLowerSetTopology.lean (hunk 1 above)
Comment (mans0954) [reply Mathlib/Topology/Order/UpperLowerSetTopology.lean]: Hopefully fixed now?

**Comment by ADedecker** [comment]
I'm extremely sorry and embarrassed that I missed all of these naming issues in #5672. It was one of the first PRs I merged directly without anyone having a look afterwards, and I clearly wasn't careful enough. Sorry again.

**Comment by ADedecker** [comment]
Could you also rename [toOrderDualHomeomorph](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Order/LowerUpperTopology.html#toOrderDualHomeomorph) to `WithLowerTopology.toDualHomeomorph` and [UpperLowerSet_toOrderDualHomeomorph](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Order/UpperLowerSetTopology.html#UpperLowerSet_toOrderDualHomeomorph) to `WithUpperSetTopology.toDualHomeomorph`?

**Comment by eric-wieser** [comment]
> I'm extremely sorry and embarrassed that I missed all of these naming issues in #5672.

Sounds like we need a linter to catch this!

**Comment by mans0954** [comment]
> Could you also rename [toOrderDualHomeomorph](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Order/LowerUpperTopology.html#toOrderDualHomeomorph) to `WithLowerTopology.toDualHomeomorph` and [UpperLowerSet_toOrderDualHomeomorph](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Order/UpperLowerSetTopology.html#UpperLowerSet_toOrderDualHomeomorph) to `WithUpperSetTopology.toDualHomeomorph`?

Done.

**Comment by mans0954** [comment]
@ADedecker @eric-wieser is there anything further that needs doing to this please?

**Pull request closed without merging**